### PR TITLE
Detect service labels for PRs based on changed files

### PR DESCRIPTION
**Pull request by melinath**
b/330551693

Some schema changes can't be detected by file changes; some file changes don't show up as schema changes. By doing both we get more accurate coverage of impacted services.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by melinath** [comment]
Looks like this isn't quite working

**Comment by roaks3** [comment]
@melinath just checking, is this ready for review yet?

**Comment by melinath** [comment]
@roaks3 yes, this is ready for review.

**Review by roaks3 — APPROVED** [review]
LGTM overall, just a few minor comments

File: .ci/magician/cmd/generate_comment.go [hunk 1]
```diff
@@ -273,6 +288,30 @@ func execGenerateComment(prNumber int, ghTokenMagicModules, buildId, buildStep,
 	sort.Strings(breakingChangesSlice)
 	data.BreakingChanges = breakingChangesSlice
 
+	// Compute additional service labels based on changed files
+	regexpLabels, err := labeler.BuildRegexLabels(labeler.EnrolledTeamsYaml)
+	affectedResources := map[string]struct{}{}
+	if err != nil {
+		fmt.Println("error building regexp labels: ", err)
+		errors["Other"] = append(errors["Other"], "Failed to parse service label mapping")
+	} else {
```
[the remark this answers is not in the document]
Comment (melinath) [reply .ci/magician/cmd/generate_comment.go:297]: hmm.... yeah I think this structure is left over from an earlier iteration of this PR and doesn't really make sense any more. I'll clean it up.

File: .ci/magician/cmd/generate_comment.go [hunk 2]
```diff
@@ -187,31 +189,41 @@ func execGenerateComment(prNumber int, ghTokenMagicModules, buildId, buildStep,
...
 		if !repo.Cloned {
 			fmt.Println("Skipping diff; repo failed to clone: ", repo.Name)
 			continue
 		}
-		diffStats, err := computeDiff(&repo, oldBranch, ctlr)
+		shortStat, err := ctlr.DiffShortStat(repo, oldBranch, newBranch)
 		if err != nil {
-			fmt.Println("diffing repo: ", err)
-			errors[repo.Title] = append(errors[repo.Title], "Failed to compute repo diff stats")
+			fmt.Println("diffing repo shortstats: ", err)
```
[the remark this answers is not in the document]
Comment (melinath) [reply .ci/magician/cmd/generate_comment.go]: resolved in fef063e

File: .ci/magician/cmd/generate_comment.go [hunk 3]
```diff
@@ -230,9 +242,12 @@ func execGenerateComment(prNumber int, ghTokenMagicModules, buildId, buildStep,
 	}
 	for _, repo := range []source.Repo{tpgRepo, tpgbRepo} {
 		if !repo.Cloned {
-			fmt.Println("Skipping breaking changes; repo failed to clone: ", repo.Name)
+			fmt.Println("Skipping diff processor; repo failed to clone: ", repo.Name)
 			continue
 		}
+		if len(repo.ChangedFiles) == 0 {
+			fmt.Println("Skipping diff processor; no diff: ", repo.Name)
```
[the remark this answers is not in the document]
Comment (melinath) [reply .ci/magician/cmd/generate_comment.go:249]: I think it should, yeah... added in 19df793

File: .ci/magician/cmd/generate_comment.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (melinath) [reply .ci/magician/cmd/generate_comment.go]: Cleaned up in 4323821 and f8c31cb.

**Review by roaks3 — APPROVED** [review]

**Pull request merged**